lib/events/meal-events.js

```diff
@@ -20,12 +24,40 @@ const removeFoodFromMeal = (e) => {
   e.preventDefault();
   meal.removeFood(food_id).
       then( (newFooter) => {
-        $(`tr[data-food_id=${food_id}]`).remove();
-        $(`table[data-id=${meal_id}] tfoot`)[0].remove();
-        $(`table[data-id=${meal_id}]`).append(`${newFooter}`);
-        $("#mealTotals").remove();
-        addMealTotals();
+        $(e.target).parent().parent().remove();
+        updateFooter(meal_id, newFooter);
+      });
+}
+
+const setAllChecks = (checked) => {
+  $('#foods-table input:checkbox').each((i,item) => {
+    $(item).prop('checked', checked);
+  })
+}
+
+const addFoodstoMeal = (e) => {
+  e.preventDefault();
+  const $mealTable = $(`table[data-name="${e.target.innerText}"]`)[0],
+        meal_id = $mealTable.dataset.id,
+        meal = meals[meal_id - 1];
+
+  const food_ids = $('#foods-table input:checkbox:checked').map( function(){
+    return $(this).val();
+  });
+
+  for(let i = 0; i < food_ids.length; i++) {
```

Comment:
Not trying to be nit picky but have you considered just writing this as ```food_ids.forEach((food_id) => { ``` ? Writing it this way is more declarative. Also, it is more "functional" and does not introduce any possible side effects because you don't need to instantiate the variable i. I have just been trying to understand what makes JS code quality better and I learned that using for loops is an imperative style of programming.